File: snippets/lesson-8-error.ts
import { of, throwError, Observable } from 'rxjs';
import { catchError } from 'rxjs/operators';

const fragileStream$: Observable<string> = throwError(() => new Error('Network Failure!'));

const safeStream$: Observable<string> = fragileStream$.pipe(
  catchError((err: Error) => {
    console.warn(`Caught: ${err.message}`);
    return of('Fallback Data');
  }),
);

safeStream$.subscribe((data: string) => console.log(data));

File: snippets/lesson-8-error.cs
using System;
using System.Reactive.Linq;

IObservable<string> fragileStream = Observable.Throw<string>(new Exception("Network Failure!"));

IObservable<string> safeStream = fragileStream.Catch<Exception>(err =>
{
    Console.WriteLine($"Caught: {err.Message}");
    return Observable.Return("Fallback Data");
});

safeStream.Subscribe((string data) => Console.WriteLine(data));